aten/src/ATen/cpu/vec256/vec256_int.h

```diff
@@ -90,15 +100,23 @@ struct Vec256<int16_t> : public Vec256i {
     vec.load(ptr);
     return vec;
   }
-  void load_partial(const void *ptr, int count) {
-    int16_t tmp_values[size];
-    std::memcpy(tmp_values, ptr, count * sizeof(int16_t));
-    load(tmp_values);
+  void load(const void *ptr, int count = size) {
+    if (count != size) {
+      int16_t tmp_values[size];
+      std::memcpy(tmp_values, ptr, count * sizeof(int16_t));
+      values = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(tmp_values));
+    } else {
+      values = _mm256_loadu_si256(reinterpret_cast<const __m256i*>(ptr));
+    }
   }
-  void store_partial(void* ptr, int count) const {
-    __at_align32__ int16_t tmp_values[size];
-    store(tmp_values);
-    std::memcpy(ptr, tmp_values, count * sizeof(int16_t));
+  void store(void* ptr, int count = size) const {
+    if (count == size) {
+      _mm256_storeu_si256(reinterpret_cast<__m256i*>(ptr), values);
+    } else {
+      __at_align32__ int16_t tmp_values[size];
+      _mm256_storeu_si256(reinterpret_cast<__m256i*>(tmp_values), values);
+      std::memcpy(ptr, tmp_values, count * sizeof(int16_t));
```

Comment:
Uh it would be nice if we could avoid having to reinstantiate those Vec structs for every type manually. It should be possible to achieve using some traits that dispatch to correct intrinsics based on the type.

Reply:
Yeah that'd be nice to have indeed.